src/index.ts

```diff
@@ -99,273 +100,186 @@ async function deletePostsInteractive(
-        // Process updates for each 'Canonical' job unless a "clone-from" argument is passed
-        const clonedJobPosts = await job.clonePost(
+    const job = new Job(page, spinner);
+
+    let jobID;
+    let jobInfo: JobInfo;
+    let regionNames = regionsArg;
+    let cloneFrom;
+
+    if (isInteractive) {
+        const { name, id } = await getJobInteractive(
+            job,
+            "What job would you like to create job posts for?",
+            spinner
+        );
+        if (!id) throw new Error(`Job cannot be found with id ${id}.`);
+        jobID = id;
+        spinner.start(`Fetching job posts for ${name}.`);
+        jobInfo = await job.getJobData(jobID);
+        if (!jobInfo.posts.length)
+            throw new Error(`Job posts cannot be found for ${jobInfo.name}.`);
+        spinner.succeed();
+
+        const jobPostID = await getJobPostInteractive(
             jobInfo.posts,
-            regionNames,
-            cloneFrom
+            "What job post should be copied?"
         );
-        // Mark all newly added job posts as live
-        const processedJobCount = await job.markAsLive(jobID, jobInfo.posts);
-
-        console.log(
-            green("✔"),
-            `${processedJobCount} job posts for ${clonedJobPosts
-                .map((post) => post.name)
-                .reduce(
-                    (previousValue, currentValue) =>
-                        previousValue + ", " + currentValue
-                )} of ${jobInfo.name} were created in ${regionNames}`
+        cloneFrom = jobPostID;
+
+        regionNames = await getRegionsInteractive(
+            "What region should those job posts be? Use space to make a selection."
         );
-        console.log("Happy hiring!");
-    } catch (error) {
-        displayError(<Error>error, spinner);
-    } finally {
-        spinner.stop();
-        currentBrowser?.close();
+
+        await deletePostsInteractive(job, jobInfo, regionNames, cloneFrom);
+    } else {
+        if (!postIDArg) throw new UserError(`Job post ID argument is missing.`);
+        if (!regionNames) throw new UserError(`Region parameter is missing.`);
+        cloneFrom = postIDArg;
+
+        spinner.start(`Fetching the job information.`);
+        jobID = await job.getJobIDFromPost(postIDArg);
+        jobInfo = await job.getJobData(jobID);
+        if (!jobInfo.posts.length)
+            throw new Error(`Job posts cannot be found for ${jobInfo.name}.`);
+        spinner.succeed();
     }
+    // Process updates for each 'Canonical' job unless a "clone-from" argument is passed
+    const clonedJobPosts = await job.clonePost(
+        jobInfo.posts,
+        regionNames,
+        cloneFrom
+    );
+    // Mark all newly added job posts as live
+    const processedJobCount = await job.markAsLive(jobID, jobInfo.posts);
+
+    console.log(
+        green("✔"),
+        `${processedJobCount} job posts for ${clonedJobPosts
+            .map((post) => post.name)
+            .reduce(
```

Comment:
this should be replaced with `join` as reduce as I think `reduce` a complex function to read and `join` does the job very well in this case

Reply:
:+1: